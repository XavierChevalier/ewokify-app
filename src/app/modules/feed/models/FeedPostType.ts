export enum AvailableFeedPostType {
  newSingle = 'newSingle',
  newAlbum = 'newAlbum',
}
