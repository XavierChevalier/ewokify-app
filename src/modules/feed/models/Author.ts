import { z } from 'zod'

export const Author = z.strictObject({
  name: z.string(),
  avatar: z.string().url(),
})

export type Author = z.infer<typeof Author>
