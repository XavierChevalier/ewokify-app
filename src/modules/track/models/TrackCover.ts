import { z } from 'zod'
import { TrackBands, TrackTitle } from '@/modules/track/models/Track'

export const TrackCover = z.string().url()
export type TrackCover = z.infer<typeof TrackCover>

export const TrackOfTrackCover = z.strictObject({
  cover: TrackCover,
  title: TrackTitle,
  bands: TrackBands,
})
export type TrackOfTrackCover = z.infer<typeof TrackOfTrackCover>
