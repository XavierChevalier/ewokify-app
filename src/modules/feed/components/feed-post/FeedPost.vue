<script setup lang="ts">
  import { computed, PropType } from 'vue'
  import { localeFormatDistance } from '@/tools/date/Date'
  import { Author } from '@/modules/feed/models/Author'
  import UserAvatar from '@/components/user-avatar/UserAvatar.vue'
  import LikeIcon from '@/components/like-icon/LikeIcon.vue'
  import CommentIcon from '@/components/comment-icon/CommentIcon.vue'

  const props = defineProps({
    author: {
      type: Object as PropType<Author>,
      required: true,
      // eslint-disable-next-line vue/valid-define-props
      validator: (value) => Author.safeParse(value).success,
    },
    content: {
      type: String,
      required: true,
    },
    numberOfLikes: {
      type: Number,
      required: true,
    },
    numberOfComments: {
      type: Number,
      required: true,
    },
    publicationDate: {
      type: Date,
      required: true,
    },
  })

  const publicationDateDistance = computed(() =>
    localeFormatDistance(props.publicationDate, new Date())
  )
</script>

<template>
  <div class="bg-white rounded p-5">
    <div class="flex items-center">
      <UserAvatar :user-name="author.name" :url="author.avatar" />

      <div class="flex flex-col pl-5">
        <span>{{ author.name }}</span>
        <span class="text-neutral">{{ publicationDateDistance }}</span>
      </div>
    </div>

    <p class="mt-5">{{ content }}</p>

    <div v-if="!!$slots.default" class="mt-2">
      <slot />
    </div>

    <div class="flex mt-5 justify-end">
      <LikeIcon show-counter :counter="numberOfLikes" />
      <CommentIcon show-counter :counter="numberOfComments" class="ml-5" />
    </div>
  </div>
</template>
