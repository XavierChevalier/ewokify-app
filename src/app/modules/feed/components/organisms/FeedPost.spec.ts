import * as Stories from '@/app/modules/feed/components/organisms/FeedPost.stories'
import { Author } from '@/app/modules/feed/models/Author'
import { omit } from 'lodash-es'
import { StorybookTestsGenerator } from '@/tests/tests-generators/StorybookTestsGenerator'

describe('FeedPost', () => {
  StorybookTestsGenerator.fromStoriesExports(Stories).property<Author>(
    'author',
    (property) => {
      const author = { name: 'Test', avatar: 'https://picsum.photos/50' }
      property
        .itShouldBeDefined()
        .itShouldHaveAValidatorFunction()
        .itShouldSuccessWith(author)
        .itShouldFailWith('missing name', omit(author, ['name']))
        .itShouldFailWith('a null avatar', { name: 'Test', avatar: null })
        .itShouldFailWith('a null name and avatar', {
          name: null,
          avatar: null,
        })
    }
  )
})
