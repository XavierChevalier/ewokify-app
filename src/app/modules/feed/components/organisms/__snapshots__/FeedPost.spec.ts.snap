// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedPost [snapshot] should render the "Default" component correctly 1`] = `
<div
  class="p-5 bg-white rounded group"
>
  <div
    class="flex items-center"
  >
    <anonymous-stub
      url="https://picsum.photos/50"
      username="TheBaronless"
    />
    <div
      class="flex flex-col pl-3"
    >
      <span>
        TheBaronless
      </span>
      <span
        class="text-neutral"
      >
        less than a minute ago
      </span>
    </div>
    <anonymous-stub
      class="ml-auto opacity-0 transition-opacity text-neutral group-hover:opacity-100"
      path="M12,16A2,2 0 0,1 14,18A2,2 0 0,1 12,20A2,2 0 0,1 10,18A2,2 0 0,1 12,16M12,10A2,2 0 0,1 14,12A2,2 0 0,1 12,14A2,2 0 0,1 10,12A2,2 0 0,1 12,10M12,4A2,2 0 0,1 14,6A2,2 0 0,1 12,8A2,2 0 0,1 10,6A2,2 0 0,1 12,4Z"
    />
  </div>
  <p
    class="mt-5"
  >
    Salut à tous, j'ai découvert un groupe complètement ouf, dites-moi ce que vous en pensez !
  </p>
  <!--v-if-->
  <div
    class="flex justify-end mt-5"
  >
    <anonymous-stub
      counter="10"
      liked="false"
      showcounter="true"
    />
    <anonymous-stub
      class="ml-5"
      counter="15"
      showcounter="true"
    />
  </div>
</div>
`;

exports[`FeedPost [snapshot] should render the "WithTrack" component correctly 1`] = `
<div
  class="p-5 bg-white rounded group"
>
  <div
    class="flex items-center"
  >
    <anonymous-stub
      url="https://picsum.photos/50"
      username="TheBaronless"
    />
    <div
      class="flex flex-col pl-3"
    >
      <span>
        TheBaronless
      </span>
      <span
        class="text-neutral"
      >
        less than a minute ago
      </span>
    </div>
    <anonymous-stub
      class="ml-auto opacity-0 transition-opacity text-neutral group-hover:opacity-100"
      path="M12,16A2,2 0 0,1 14,18A2,2 0 0,1 12,20A2,2 0 0,1 10,18A2,2 0 0,1 12,16M12,10A2,2 0 0,1 14,12A2,2 0 0,1 12,14A2,2 0 0,1 10,12A2,2 0 0,1 12,10M12,4A2,2 0 0,1 14,6A2,2 0 0,1 12,8A2,2 0 0,1 10,6A2,2 0 0,1 12,4Z"
    />
  </div>
  <p
    class="mt-5"
  >
    Salut à tous, j'ai découvert un groupe complètement ouf, dites-moi ce que vous en pensez !
  </p>
  <div
    class="mt-2"
  >
    
    <div
      class="bg-white-dark rounded p-2 flex"
    >
      <img
        alt="The track cover"
        class="rounded w-16 h-16"
        src="https://picsum.photos/50?1"
      />
      <div
        class="flex flex-col ml-2 leading-tight justify-center"
      >
        <span
          class="font-bold"
        >
          G.O.A.T.
        </span>
        
        <span>
          Polyphia
        </span>
        
      </div>
      <div
        class="flex self-center ml-auto"
      >
        <anonymous-stub
          class="text-black-light"
          counter="0"
          liked="false"
          showcounter="false"
        />
        <span
          class="ml-3"
        >
          2:30
        </span>
      </div>
    </div>
    
  </div>
  <div
    class="flex justify-end mt-5"
  >
    <anonymous-stub
      counter="10"
      liked="false"
      showcounter="true"
    />
    <anonymous-stub
      class="ml-5"
      counter="15"
      showcounter="true"
    />
  </div>
</div>
`;
